public/views/main.js

```diff
@@ -284,14 +284,14 @@ function onFrame(event, debug) {
 		}
 	}
 
-	// check for score change
-	if (--pending_line === 0) {
+	// check for score change or score stayed at 999999 but the line count changed.
+	if (--pending_line === 0 || (diff.cleared_lines > 0 && transformed.score == 999999)) {
```

Comment:
Is this check required here? Since the same check appears is below

If it's here then it could go the if statement nside when pendingline is !== 1 which could be disastrous as you could read the wrong amount of lines cleared.

Reply:
Thank you for this I'll take a look when I'm home